backend/rdp/peer.c

```diff
@@ -105,6 +105,7 @@ static int xf_input_mouse_event(rdpInput *input,
 		struct wlr_event_pointer_motion_absolute event = { 0 };
 		event.device = wlr_device;
 		event.time_msec = timespec_to_msec(&now);
+		event.time_frac_nsec = now.tv_nsec;
```

Comment:
No modulo here?

Reply:
Valid timespec objects should have the correct range [0-999999999] in the tv_nsec field already, unless I'm wrong.

Reply:
Oh right, indeed